dataflux_pytorch/lightning/dataflux_lightning_checkpoint.py

```diff
@@ -52,8 +52,7 @@ def _parse_gcs_path(self, path: Union[str, Path]) -> str:
             bucket_name, prefix = split
         if not bucket_name:
             raise ValueError("Bucket name must be non-empty")
-        if not self.bucket:
-            self.bucket = self.storage_client.bucket(bucket_name)
+        self.bucket = self.storage_client.bucket(bucket_name)
```

Comment:
I wonder if this could cause problems in any sort of multi-threaded scenario, if one thread calls _parse_gcs_path, then another thread calls _parse_gcs_path, then the original thread makes its actual API call.

To avoid any problems like that, would it work to initialize it wherever self.bucket is used today?

Reply:
As discussed offline initializing bucket client where we need it.